src/node_file.cc

```diff
@@ -559,6 +560,9 @@ static void InternalModuleStat(const FunctionCallbackInfo<Value>& args) {
   if (rc == 0) {
     const uv_stat_t* const s = static_cast<const uv_stat_t*>(req.ptr);
     rc = !!(s->st_mode & S_IFDIR);
+    if (rc == 0 && (s->st_mode & (S_IFIFO | S_IFSOCK))) {
```

Comment:
I am not 100% convinced this is correct.  Is there a reason you left out S_IFCHR?

My gut instinct is that the logic should look something like this:
```c
if (S_ISREG(s->st_mode))
  rc = 0;
else if (S_ISDIR(s->st_mode))
  rc = 1;
else
  rc = 2;
```
(Whether that compiles on Windows is left as an exercise to the reader.)